models/templates.go

```diff
@@ -8,15 +8,25 @@ import (
 )
 
 type Templates struct {
-	templates *template.Template
+	templates map[string]*template.Template
 }
 
 func (t *Templates) Render(w io.Writer, name string, data interface{}, c echo.Context) error {
-	return t.templates.ExecuteTemplate(w, name, data)
+	return t.templates[name].Execute(w, data)
```

Comment:
lets add some error handling here with a useful message so its easy to debug if we mess up with the key